test/integration/RoutePatternTest.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const TestHelper = require('../TestHelper');
+const mockyeah = TestHelper.mockyeah;
+const request = TestHelper.request;
+
+describe('Route Patterns', () => {
+  it('should work with route parameter', (done) => {
+    mockyeah.get('/service/:key');
+
+    request
+      .get('/service/exists')
+      .expect(200, done);
+  });
+
+  it('should work with regular expression slash any count', (done) => {
+    mockyeah.get('/service/(.{0,})');
+
+    request
+      .get('/service/exists')
+      .expect(200, done);
+  });
+
+  it('should work with regular expression slash any star', (done) => {
+    mockyeah.get('/(.*)');
+
+    request
+      .get('/service/exists')
+      .expect(200, done);
+  });
+
+  it('should work with regular expression any star', (done) => {
+    mockyeah.get('(.*)');
+
+    request
+      .get('/service/exists')
+      .expect(200, done);
+  });
+
+  it('should work with star', (done) => {
+    mockyeah.get('*');
+
+    request
+      .get('/service/exists')
+      .expect(200, done);
+  });
```

Comment:
This is the behavior you are intending to add correct? I believe all the other tests assert existing behavior.

Reply:
... might have spoke to soon, currently knee deep in your `RouteResolver` changes.

Reply:
@ryanricard I'm just proving here that Express route matching still works. Since we're no long mounting each route directly with Express, but matching manually after parsing with `path-to-regexp`, which follows Express path matching semantics, I wanted to ensure backwards compatibility.

Reply:
To be clear, there are no intentional new features with this pull request. It's merely a refactoring to support new features in the future.